frontend/components/containers/comments/CommentList.vue

```diff
@@ -0,0 +1,143 @@
+    </template>
+    <template v-slot:top>
+      <v-text-field
+        v-model="search"
+        prepend-inner-icon="search"
+        :label="$t('generic.search')"
+        single-line
+        hide-details
+        filled
+      />
+    </template>
+    <template v-slot:item.action="{ item }">
+      <v-btn
+        small
+        color="primary text-capitalize"
+        @click="toLabeling(item)"
+      >
+        {{ $t('comments.commentView') }}
+      </v-btn>
+    </template>
+  </v-data-table>
+</template>
+
+<script>
+import Vue from 'vue'
+import { mapActions, mapGetters, mapState, mapMutations } from 'vuex'
+import VueFilterDateFormat from '@vuejs-community/vue-filter-date-format'
+import VueFilterDateParse from '@vuejs-community/vue-filter-date-parse'
+Vue.use(VueFilterDateFormat)
+Vue.use(VueFilterDateParse)
+
+export default {
+  async fetch() {
+    await this.getDocumentList({
+      projectId: this.$route.params.id,
+      ...this.$route.query
+    })
+    await this.getCommentListComplete({
+      projectId: this.$route.params.id
+    })
+  },
+  data() {
+    return {
+      search: this.$route.query.q,
+      options: {},
+      headers: [
+        {
+          text: 'Created At',
+          align: 'left',
+          value: 'created_at'
+        },
+        {
+          text: 'Document',
+          value: 'document_text'
+        },
+        {
+          text: 'Username',
+          value: 'username'
+        },
+        {
+          text: 'Text',
+          value: 'text'
+        },
+        {
+          text: this.$t('dataset.action'),
+          align: 'left',
+          value: 'action',
+          sortable: false
+        }
+      ]
+    }
+  },
+  computed: {
+    ...mapState('documents', ['items', 'comments', 'loading', 'selectedComments']),
+    ...mapGetters('projects', ['getLink'])
+  },
+  watch: {
+    '$route.query': '$fetch',
+    options: {
+      handler(newvalue, oldvalue) {
+        this.$router.push({
+          query: {
+            limit: this.options.itemsPerPage,
+            offset: (this.options.page - 1) * this.options.itemsPerPage,
+            q: this.search
+          }
+        })
+      },
+      deep: true
+    },
+    search() {
+      this.$router.push({
+        query: {
+          limit: this.options.itemsPerPage,
+          offset: 0,
+          q: this.search
+        }
+      })
+      this.options.page = 1
+    }
+  },
+  methods: {
+    ...mapActions('documents', ['getCommentListComplete', 'getDocumentList']),
+    ...mapMutations('documents', ['updateSelectedComments']),
+
+    toLabeling(comment) {
+      const index = this.items.findIndex(item => item.id === comment.document)
+      const offset = (this.options.page - 1) * this.options.itemsPerPage
+      const page = offset + index + 1
+      this.$router.push({
+        path: this.localePath(`/projects/${this.$route.params.id}/${this.getLink}`),
+        query: {
+          page,
+          q: this.search
```

Comment:
search text should not be passed here.

Reply:
I took this method from the DocumentList.vue. What is the use of the q-parameter?

Reply:
it's the search query. used in combination with the page to get to the right document in the annotation view. useful if you want to annotate documents that contain a certain string.